metrics/influx/influx_test.go

```diff
@@ -31,6 +30,33 @@ func TestCounter(t *testing.T) {
 	}
 }
 
+func ExampleCounter() {
+	in := New(map[string]string{"a": "b"}, influxdb.BatchPointsConfig{}, log.NewNopLogger())
+	counter := in.NewCounter("influx_counter")
+	counter.Add(10)
+	counter.With("error", "true").Add(1)
+	counter.With("error", "false").Add(2)
+	counter.Add(50)
+
+	client := &bufWriter{}
+	in.WriteTo(client)
+
+	expectedLines := []string{
+		`(influx_counter,a=b count=60) [0-9]{19}`,
+		`(influx_counter,a=b,error=true count=1) [0-9]{19}`,
+		`(influx_counter,a=b,error=false count=2) [0-9]{19}`,
+	}
+
+	if err := extractAndPrintMessage(expectedLines, client.buf.String()); err != nil {
+		fmt.Println(err.Error())
+	}
+
+	// Output:
+	// influx_counter,a=b count=60
+	// influx_counter,a=b,error=true count=1
+	// influx_counter,a=b,error=false count=2
+}
```

Comment:
Since this is now an Example, the test runner will take care of validating the output; there's no need for a separate step with the regexes. So,

```go
in := New(...)
counter := in.NewCounter("...")
// etc.

client := &bufWriter{}
in.WriteTo(client)
io.Copy(os.Stdout, client.buf)

// Output:
// foo
// bar
// baz
```

should be sufficient.